core/src/main/scala/akka/kafka/ProducerMessage.scala

```diff
@@ -212,11 +212,28 @@ object ProducerMessage {
     def passThrough: PassThrough = message.passThrough
   }
 
+  object Result {
+    def apply[K, V, PassThrough](metadata: RecordMetadata,
+                                 message: Message[K, V, PassThrough]): Result[K, V, PassThrough] =
+      new Result(metadata, message)
+  }
+
   final case class MultiResultPart[K, V] private (
       metadata: RecordMetadata,
       record: ProducerRecord[K, V]
   )
 
+  object MultiResultPart {
+
+    /**
+     * Provide access to `apply` which Scala3 makes private.
+     */
+    def create[K, V](
```

Comment:
why is this `create` and `Result` has `apply`?

Reply:
Do I understand it correctly that Scala 3 doesn't add `apply` if the constructor is private? Can we still name it `apply` to be source compatible, or is that in conflict with Scala 2?

Reply:
I think Scala3 makes the constructor private, which is why my suggestion is to provide `apply` through the companion object which has access to private methods (the constructor).

The `create` approach also works, but we should stick to `apply`. I missed that and made it consistent now.